File: 1395-count-number-of-teams/1395-count-number-of-teams.kt
class Solution {
    fun numTeams(rating: IntArray): Int {
        val n = rating.size
        val up = Array(n) {0}
        val down = Array(n) {0}
        var res = 0
        for (i in 0..n-1) {
            for (j in 0..i-1) {
                if (rating[j] < rating[i]) {
                    up[i]++
                    res += up[j]
                } else {
                    down[i]++
                    res += down[j]
                }
            }
        }
        return res
    }
}

File: 1395-count-number-of-teams/1395-count-number-of-teams.py
class Solution:
    def numTeams(self, rating: List[int]) -> int:
        n = len(rating)
        up, down, res = [0] * n, [0] * n, 0
        for i in range(0, n):
            for j in range(0, i):
                if rating[j] < rating[i]:
                    up[i] += 1
                    res += up[j]
                else:
                    down[i] += 1
                    res += down[j]
        return res
                    
        
    